### ISSUE TITLE
Wrong QC display for rootlets segmentation

### ISSUE THREAD

**Issue by sandrinebedard** [spinalcordtoolbox/spinalcordtoolbox#4716]
### Description
When generating the QC report for rootlets with a T2w image of `0.4x0.4x0.8 mm`,  the spinal cord is squeezed in the R-L direction:

![Capture d’écran, le 2024-11-26 à 14 00 30](https://github.com/user-attachments/assets/5d8ff22b-3180-4b9c-a2f8-f512cb4be7c6)

In FSLeyes the axial slice looks like:

![image](https://github.com/user-attachments/assets/8e697af8-d2b8-47ad-bc94-deaf6ec38a8b)
### Steps to Reproduce

Command:
~~~
sct_deepseg -i ${file}.nii.gz -task seg_spinal_rootlets_t2w -qc ${PATH_QC} -o ${FILELABEL}.nii.gz
~~~
Data:
[Uploading t2.nii.gz…]()

**Comment by jcohenadad** [comment]
>  the QC function for the rootlets is separate, this may explain!

right-- I suspect there is no isotropic interpolation done for this QC

**Comment by joshuacwnewton** [comment]
> right-- I suspect there is no isotropic interpolation done for this QC (where it should)

Indeed! 

For comparison, here is the correct (**non-rootlets**) `sct_deepseg` axial QC (isotropic resampling):

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/89b1ffcbc649cb0f90573cb37aa378f1a3cd3b08/spinalcordtoolbox/reports/qc2.py#L316-L323

However, the rootlets QC skips resampling, because the thin (often single-voxel) rootlets labels may not be displayed accurately. We try to display the voxels faithfully, and thus the anisotropic resolution is visible.

I think that rather than fixing this via resampling, we can fix this via matplotlib's options for `imshow`. (i.e. we still preserve the voxels, but we change how we display the image array). I've got a prototype I'll open a PR for:

![image](https://github.com/user-attachments/assets/5c82895f-14cb-4520-ae7b-2ee84c520c26)

(The example data @sandrinebedard uploaded seems to have not been uploaded correctly (GitHub glitch?), so I took `sct_testing_data/t2/t2.nii.gz`, resampled it to `0.4x0.4x0.8mm`, then tweaked the QC report so that it would display correctly.)
